src/testing/order_test.rs

```diff
@@ -0,0 +1,412 @@
+        // simulate transaction being submitted to the builder
+        for res in handle_received_txns(
+            &senders.transactions,
+            all_transactions[round].clone(),
+            TransactionSource::HotShot,
+        )
+        .await
+        {
+            res.unwrap();
+        }
+
+        // get transactions submitted in previous rounds, [] for genesis
+        // and simulate the block built from those
+        let transactions = prev_proposed_transactions.take().unwrap_or_default();
+        let (quorum_proposal, quorum_proposal_msg, da_proposal_msg, builder_state_id) =
+            calc_proposal_msg(NUM_STORAGE_NODES, round, prev_quorum_proposal, transactions).await;
+
+        prev_quorum_proposal = Some(quorum_proposal.clone());
+
+        // send quorum and DA proposals for this round
+        senders
+            .da_proposal
+            .broadcast(da_proposal_msg)
+            .await
+            .unwrap();
+        senders
+            .quorum_proposal
+            .broadcast(quorum_proposal_msg)
+            .await
+            .unwrap();
+
+        let req_msg = get_req_msg(round as u64, builder_state_id).await;
+
+        // give builder state time to fork
+        async_sleep(Duration::from_millis(100)).await;
+
+        // get the builder state for parent view we've just simulated
+        global_state
+            .read_arc()
+            .await
+            .spawned_builder_states
+            .get(&req_msg.1)
+            .expect("Failed to get channel for matching builder")
+            .broadcast(req_msg.2.clone())
+            .await
+            .unwrap();
+
+        // get response
+        // in the next round we will use received transactions to simulate
+        // the block being proposed
+        let res_msg = req_msg
+            .0
+            .recv()
+            .timeout(Duration::from_secs(10))
+            .await
+            .unwrap()
+            .unwrap();
+
+        // play with transactions propsed by proposers: skip the whole round OR interspersed some txs randomly OR remove some txs randomly
+        if let MessageType::<TestTypes>::RequestMessage(ref request) = req_msg.2 {
+            let view_number = request.requested_view_number;
+            if view_number == ViewNumber::new(skip_round as u64) {
+                prev_proposed_transactions = None;
+            } else {
+                let mut proposed_transactions = res_msg.transactions.clone();
+                if view_number == ViewNumber::new(adjust_add_round as u64) {
+                    proposed_transactions.insert(
+                        rand::random::<usize>() % NUM_TXNS_PER_ROUND,
+                        TestTransaction::new(vec![
+                            adjust_add_round as u8,
+                            (NUM_TXNS_PER_ROUND + 1) as u8,
+                        ]),
+                    );
+                } else if view_number == ViewNumber::new(adjust_remove_round as u64) {
```

Comment:
Yeah originally `adjust_remove_round` is set to a random number less than `NUM_ROUNDS` so that the test is expected to fail (because we skip a middle part of the bundle). But since the test will fail after the parameter enabled, now it is disabled by setting to `NUM_ROUNDS + 1`.